packages/stripe/src/lib/stripe.module.ts

```diff
@@ -39,34 +39,49 @@ const makeController = (prefix: string, route: string) => {
   return Controller;
 };
 
-export const stripeModule = new AppModule(
+export class StripeModule extends createModule(
   {
     config: stripeModuleConfig,
-    providers: [
-      {
-        provide: 'Stripe',
-        deps: [stripeModuleConfig.all()],
-        useFactory: (config: typeof stripeModuleConfig.type) => {
-          const { apiKey } = config;
-          return new Stripe(apiKey, {
-            typescript: true,
-            apiVersion: '2020-08-27',
-          });
-        },
-      },
-    ],
-    exports: ['Stripe'],
+    exports: [Stripe],
   },
   'stripe'
-).setup((stripeModule, config) => {
-  if (config.webhookConfig) {
-    const webhookConfig = config.webhookConfig!;
-    stripeModule.addProvider(StripeWebhookService);
-    stripeModule.addController(
-      makeController(
-        webhookConfig.controllerRoutePrefix!,
-        webhookConfig.controllerRoute!
-      )
-    );
+) {
+
+  process() {
+    this.addProvider({
+      provide: Stripe,
+      useFactory: () => {
+        return new Stripe(this.config.apiKey, {
+          typescript: true,
+          apiVersion: '2020-08-27',
+        });
+      },
+    });
+
+    if (this.config.webhookConfig) {
+      this.addProvider(StripeWebhookService);
+      this.addController(
+        makeController(
+          this.config.webhookConfig.controllerRoutePrefix,
+          this.config.webhookConfig.controllerRoute,
+          this.config.webhookConfig.secret
+        )
+      );
+    }
+  }
+
+  processController(module: AppModule<any>, controller: ClassType) {
```

Comment:
I'll probably include the ability to have handlers be on either controllers or regular providers since the handlers are usually dealing directly with business logic after receiving the event. I think users like the flexibility of being able to use either one

Reply:
You can for sure with the processProvider hook. I wouldn't do it though in my code because I believe services shouldn't receive external events. That is what controllers are (also semantically) usually used for.

Reply:
In this case its the provided StripeController which really receives the external event. The idea is that it then hands that event to the appropriate service in order for whatever logic you need to be executed. But like you said, this is a matter of preference so I definitely think I'll give people the option to use either approach and do what makes sense for them